sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/tests/ScenarioTests/RestoreDatabaseAccountOperationsTests.cs

```diff
@@ -0,0 +1,80 @@
+            using (MockContext context = MockContext.Start(this.GetType()))
+            {
+                // Create client
+                CosmosDBManagementClient cosmosDBManagementClient = CosmosDBTestUtilities.GetCosmosDBClient(context, handler1);
+                ResourceManagementClient resourcesClient = CosmosDBTestUtilities.GetResourceManagementClient(context, handler1);
+                string restoredatabaseAccountName = TestUtilities.GenerateName(prefix: "restoredaccountname");
+
+                DatabaseAccountGetResults databaseAccount = await cosmosDBManagementClient.DatabaseAccounts.GetAsync(resourceGroupName, sourceDatabaseAccountName);
+                DateTime restoreTs = DateTime.Parse(restoreTimestamp);
+
+                List<RestorableDatabaseAccountGetResult> restorableAccounts = (await cosmosDBManagementClient.RestorableDatabaseAccounts.ListAsync()).ToList();
+                RestorableDatabaseAccountGetResult restorableDatabaseAccount = restorableAccounts.
+                    SingleOrDefault(account => account.Name.Equals(databaseAccount.InstanceId, StringComparison.OrdinalIgnoreCase));
+
+                List<Location> locations = new List<Location>
+                {
+                    new Location(locationName: location)
+                };
+
+                RestoreReqeustDatabaseAccountCreateUpdateProperties databaseAccountCreateUpdateProperties = new RestoreReqeustDatabaseAccountCreateUpdateProperties
+                {
+                    Locations = locations,
+                    RestoreParameters = new RestoreParameters()
+                    {
+                        RestoreMode = "PointInTime",
+                        RestoreTimestampInUtc = restoreTs,
+                        RestoreSource = restorableDatabaseAccount.Id
+                    }
+                };
+
+                DatabaseAccountCreateUpdateParameters databaseAccountCreateUpdateParameters = new DatabaseAccountCreateUpdateParameters
+                {
+                    Location = location,
+                    Tags = new Dictionary<string, string>
+                    {
+                        {"key1","value1"},
+                        {"key2","value2"}
+                    },
+                    Kind = "GlobalDocumentDB",
+                    Properties = databaseAccountCreateUpdateProperties
+                };
+
+                DatabaseAccountGetResults restoredDatabaseAccount = (await cosmosDBManagementClient.DatabaseAccounts.CreateOrUpdateWithHttpMessagesAsync(resourceGroupName, restoredatabaseAccountName, databaseAccountCreateUpdateParameters)).Body;
+                Assert.NotNull(restoredDatabaseAccount);
+                Assert.NotNull(restoredDatabaseAccount.RestoreParameters);
+                Assert.Equal(restoredDatabaseAccount.RestoreParameters.RestoreSource.ToLower(), restorableDatabaseAccount.Id.ToLower());
+                Assert.True(restoredDatabaseAccount.BackupPolicy is ContinuousModeBackupPolicy);
```

Comment:
Is there a reason you not using Assert.Equal ?

Reply:
The backupPolicy is the base class.
For accounts with Periodic mode backup type, the derived class "PeriodicModeBackupPolicy" is returned. 
For accounts with Continuous mode backup type, the derived class "ContinuousModeBackupPolicy" is returned. 
So to assert if the restored account has continuous backup policy, we just need to check if backupPolicy is the ContinuousModeBackupPolicy,